src/mongoose_rabbit_worker.erl

```diff
@@ -132,18 +135,36 @@ publish_groupchat_msg_recv(Channel, EventData = #{to_jid := To,
 publish_chat_msg_event(Channel, RoutingKey, #{from_jid := From,
                                               to_jid := To,
                                               msg := UserMsg,
-                                              exchange := Exchange}) ->
+                                              exchange := Exchange,
+                                              host := Host}) ->
     Message = make_chat_msg(From, To, UserMsg),
-    publish(Channel, Exchange, RoutingKey, Message).
+    publish(Channel, Exchange, RoutingKey, Message, Host).
 
 -spec publish(Channel :: pid(), Exchange :: binary(), RoutingKey :: binary(),
-              Message :: binary()) -> term().
-publish(Channel, Exchange, RoutingKey, Message) ->
-    amqp_channel:call(Channel,
-                      #'basic.publish'{
-                         exchange = Exchange,
-                         routing_key = RoutingKey},
-                      #amqp_msg{payload = Message}).
+              Message :: binary(), Host :: jid:server()) -> term().
+publish(Channel, Exchange, RoutingKey, Message, Host) ->
+    {PublishTime, _} = timer:tc(amqp_channel, call,
```

Comment:
Publish could folllow a pattern present in this module that all the calls to `amqp_*` are wrapped into small functions. That would result in calling two "locals": `call_channel` and `wait_for_confirm_on_channel` (sth along these lines)